src/runtime_strings.js

```diff
@@ -118,6 +118,85 @@ function UTF8ToString(ptr, maxBytesToRead) {
 #endif
 }
 
+
```

Comment:
Please move all the added content in this file over to `src/library_strings.js`. The file `runtime_strings.js` should not receive any more content, it exists for historical reasons. All new functions can go to `library_strings.js`.

Reply:
accepted

Reply:
I tried to move code to `src/library_strings.js`, but it seems to have something to be related to the `MINIMAL_RUNTIME` flag, no matter how the unit test complains `ReferenceError: UTF8ToStringNBytes is not defined`.

The half-done commit is here https://github.com/LanderlYoung/emscripten/commit/f9149d794f38beedb8d4d6ff6c0d53ebb6cfbec6

If there is and doc related, please let me know, thanks.

Reply:
The way the `library_xxx.js` system works is that symbols are only ever included on on demand.

There are several ways a library symbol can get included:

1. By being directly referenced by native code
2. By being add to `-s  DEFAULT_LIBRARY_FUNCS_TO_INCLUDE` list.
3. Via the `__deps` entry to another symbol that is incuded.

(1) doesn't apply to JS functions that are only callable from JS such as these ones so you can ignore that.  To signal these are JS-only functions you should prefix them with `$`.

Reply:
It looks like `src/library_strings.js` is only currently included in MINIMAL_RUNTIME mode.  Its job is currently to make the functions in `runtime_strings_extra.js` available as JS library, but only MINIMAL_RUNTIME mode.

`runtime_strings_extra.js` is include directly in normal (non-MINIMAL) mode.

So some refactoring would need to be doing before these functions can be make universally available via the library system.

I'd be tempted to add all these new functions to `runtime_strings_extra.js` + `src/library_strings.js` for now.  At least then they will be used in library form under MINIMAL_RUNTIME.

The alternative is that we do some refactoring such at `src/library_strings.js` is used in all modes.